docs/source/bettertransformer/contribution/content.mdx

```diff
@@ -0,0 +1,176 @@
+            (key): Linear(in_features=768, out_features=768, bias=True)
+            (value): Linear(in_features=768, out_features=768, bias=True)
+            (dropout): Dropout(p=0.1, inplace=False)
+          )
+          (output): BertSelfOutput(
+            (dense): Linear(in_features=768, out_features=768, bias=True)
+            (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
+            (dropout): Dropout(p=0.1, inplace=False)
+          )
+        )
+        (intermediate): BertIntermediate(
+          (dense): Linear(in_features=768, out_features=3072, bias=True)
+          (intermediate_act_fn): GELUActivation()
+        )
+        (output): BertOutput(
+          (dense): Linear(in_features=3072, out_features=768, bias=True)
+          (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
+          (dropout): Dropout(p=0.1, inplace=False)
+        )
+      )
+    )
+  )
+  (pooler): BertPooler(
+    (dense): Linear(in_features=768, out_features=768, bias=True)
+    (activation): Tanh()
+  )
+)
+```
+You can clearly see that the layers that needs to be replaced are the `BertLayer` modules. 
+
+### Step 2: Building the `xxxLayerBetterTransformer` module
+
+Check that the identified module is not already copied from another module (by inspecting the source code in [`transformers`](https://github.com/huggingface/transformers) and checking that the class definition does not start with `# Copied from ...`) - and if not, create a folder in `bettertransformer/models` (here `models/bert`).
+Once there, create a file named `modeling_bert_bettertransformer.py`, and start with those lines:
+```python
+import torch
+import torch.nn as nn
+
+from ..base import BetterTransformerBaseLayer
+
+
+class BertLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, bert_layer, config):
+...
+```
+Now, make sure to fill all the necessary attributes, the list of attributes are:
+- `in_proj_weight`
+- `in_proj_bias`
+- `out_proj_weight`
+- `out_proj_bias`
+- `linear1_weight`
+- `linear1_bias`
+- `linear2_weight`
+- `linear2_bias`
+- `norm1_eps`
+- `norm1_weight`
+- `norm1_bias`
+- `norm2_weight`
+- `norm2_bias` 
+- `num_heads`
+- `embed_dim`
+Note that these attributes correspond to all the components that are necessary to run a Transformer Encoder module, check the figure 1 on the ["Attention Is All You Need"](https://arxiv.org/pdf/1706.03762.pdf) paper.
+Once you filled all these attributes (sometimes the `query`, `key` and `value` layers needs to be "contigufied", check the `modeling_bert_bettertransformer.py` file to understand more.)
+
+Make sure also to add the lines:
+```python
+self.is_last_layer = False
+self.validate_bettertransformer()
+```
+
+### Step 3: Building the forward pass
+
+First of all, start with the line `super().forward_checker()`, this is needed so that the parent class can check all the safety checkers before.
```

Comment:
```suggestion
First of all, start with the line `super().forward_checker()`, this is needed so that the parent class can run all the safety checkers before.
```